Mathlib/Analysis/InnerProductSpace/Basic.lean

```diff
@@ -307,39 +331,95 @@ theorem inner_sub_sub_self (x y : F) : ⟪x - y, x - y⟫ = ⟪x, x⟫ - ⟪x, y
+  rw [inner_smul_right, mul_comm]
+
+/-- An auxiliary equality useful to prove the **Cauchy–Schwarz inequality**. Here we use the
+standard argument involving the discriminant of quadratic form. -/
+lemma cauchy_schwarz_aux' (x y : F) (t : ℝ) : 0 ≤ seminormSqF x * t * t + 2 * re ⟪x, y⟫ * t
+    + seminormSqF y := by
+  calc 0 ≤ re ⟪(ofReal t : 𝕜) • x + y, (ofReal t : 𝕜) • x + y⟫ := inner_self_nonneg
+  _ = re (⟪(ofReal t : 𝕜) • x, (ofReal t : 𝕜) • x⟫ + ⟪(ofReal t : 𝕜) • x, y⟫
+      + ⟪y, (ofReal t : 𝕜) • x⟫ + ⟪y, y⟫) := by rw [inner_add_add_self ((ofReal t : 𝕜) • x) y]
+  _ = re ⟪(ofReal t : 𝕜) • x, (ofReal t : 𝕜) • x⟫
+      + re ⟪(ofReal t : 𝕜) • x, y⟫ + re ⟪y, (ofReal t : 𝕜) • x⟫ + re ⟪y, y⟫ := by
+      simp only [map_add]
+  _ = seminormSq x * t * t + re (⟪x, y⟫ * t) + re (⟪y, x⟫ * t) + re ⟪y, y⟫ := by rw
+    [inner_smul_smul_re, inner_smul_left_re, inner_smul_right_re]
+  _ = seminormSq x * t * t + re ⟪x, y⟫ * t + re ⟪y, x⟫ * t + re ⟪y, y⟫ := by rw [mul_comm ⟪x,y⟫ _,
+    RCLike.re_ofReal_mul, mul_comm t _, mul_comm ⟪y,x⟫ _, RCLike.re_ofReal_mul, mul_comm t _]
+  _ = seminormSq x * t * t + re ⟪x, y⟫ * t + re ⟪y, x⟫ * t + seminormSq y := by exact rfl
+  _ = seminormSq x * t * t + re ⟪x, y⟫ * t + re ⟪x, y⟫ * t + seminormSq y := by rw [inner_re_symm]
+  _ = seminormSq x * t * t + 2 * re ⟪x, y⟫ * t + seminormSq y := by ring
+
+/-- Another auxiliary equality related with the **Cauchy–Schwarz inequality**: the square of the
+seminorm of `⟪x, y⟫ • x - ⟪x, x⟫ • y` is equal to `‖x‖ ^ 2 * (‖x‖ ^ 2 * ‖y‖ ^ 2 - ‖⟪x, y⟫‖ ^ 2)`.
+We use `InnerProductSpace.ofCore.normSq x` etc (defeq to `is_R_or_C.re ⟪x, x⟫`) instead of `‖x‖ ^ 2`
 etc to avoid extra rewrites when applying it to an `InnerProductSpace`. -/
-theorem cauchy_schwarz_aux (x y : F) :
-    normSqF (⟪x, y⟫ • x - ⟪x, x⟫ • y) = normSqF x * (normSqF x * normSqF y - ‖⟪x, y⟫‖ ^ 2) := by
-  rw [← @ofReal_inj 𝕜, ofReal_normSq_eq_inner_self]
+theorem cauchy_schwarz_aux (x y : F) : seminormSqF (⟪x, y⟫ • x - ⟪x, x⟫ • y)
+    = seminormSqF x * (seminormSqF x * seminormSqF y - ‖⟪x, y⟫‖ ^ 2) := by
+  rw [← @ofReal_inj 𝕜, ofReal_seminormSq_eq_inner_self]
   simp only [inner_sub_sub_self, inner_smul_left, inner_smul_right, conj_ofReal, mul_sub, ←
-    ofReal_normSq_eq_inner_self x, ← ofReal_normSq_eq_inner_self y]
+    ofReal_seminormSq_eq_inner_self x, ← ofReal_seminormSq_eq_inner_self y]
   rw [← mul_assoc, mul_conj, RCLike.conj_mul, mul_left_comm, ← inner_conj_symm y, mul_conj]
   push_cast
   ring
 #align inner_product_space.core.cauchy_schwarz_aux InnerProductSpace.Core.cauchy_schwarz_aux
 
 /-- **Cauchy–Schwarz inequality**.
-We need this for the `Core` structure to prove the triangle inequality below when
-showing the core is a normed group.
+We need this for the `PreInnerProductSpace.Core` structure to prove the triangle inequality below
+when showing the core is a normed group and to take the quotient.
 -/
 theorem inner_mul_inner_self_le (x y : F) : ‖⟪x, y⟫‖ * ‖⟪y, x⟫‖ ≤ re ⟪x, x⟫ * re ⟪y, y⟫ := by
```

Comment:
I have a feeling there's a much easier-to-formalize proof here, but I don't have time at the moment. I'll come back to it.

Reply:
OK, I leave this as is for now.